airbyte-cdk/python/airbyte_cdk/sources/declarative/checks/check_stream.py

```diff
@@ -28,7 +28,7 @@ def __post_init__(self, parameters: Mapping[str, Any]) -> None:
         self._parameters = parameters
 
     def check_connection(self, source: Source, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
-        streams = source.streams(config)  # type: ignore # source is always a DeclarativeSource, but this parameter type adheres to the outer interface
+        streams = source.all_streams(config)  # type: ignore # source is always a DeclarativeSource, but this parameter type adheres to the outer interface
```

Comment:
Teach me your ways, what is the difference betweeen streams and all_streams? (ignore if it's in this PR, reading through it)

Reply:
good question, I mention it in a comment: https://github.com/airbytehq/airbyte/pull/46662/files#r1792816384

But the reason why we can't just rewrite the `streams()` method is because within the existing Python CDK `core.py`, when processing synchronous streams, we invoke the `streams()` method and in that context we don't want to return the concurrent streams that aren't compatible in that are of code.

Reply:
As we discussed earlier, it’s preferable to use the stream method and condition the behavior accordingly. This approach adds some complexity, but it provides a tradeoff by allowing simpler modifications later. With this setup, when the core will be able to handle concurrent streams, we’ll get a stream generation interface for free.

Reply:
yep this is addressed in my latest commit using the optional param `include_concurrent_streams`